### DISCUSSION TITLE
feature requests specifically for private-tracker quality of life

### DISCUSSION THREAD

**Discussion by ldpr** [torrust/torrust-index-archive#96] [Feature requests]
hi all - first off thanks @WarmBeer for this project, we love it. we've it deployed for 6 weeks and we're at around the 8k account mark with ~2-3k created torrents. Our users love it, thank you!

There are some QOL features we'd really love, I was going to create separate issues but figured since it's private-tracker specific it's probably better in one.

1. deleted torrents need to be de-whitelisted from the tracker. for example, if we delete a torrent to replace it with an updated version, the previous version will continue announcing fine to the tracker. this makes it a pain for us to keep our archive sane as we have to check for duplicates every few days. this would be _huge_ for us and our users.

2. user moderation/groups/roles. for example
   1. an uploader role (by default upload would be disabled without this role) we just hid the upload button from non administrators with this patch https://github.com/rpdl-net/torrust/commit/1b7aecc443f009f91f5961ac4c91035041ee52f0 but obviously, they could still use the API to upload
   2. a moderator role who can edit torrent description/title, ban users, upload and do basically everything an admin can except change site settings.

3. the ability to edit titles of torrent pages. at the moment if we make a typo we delete and remake, but it's pretty clunky (especially as deleted torrents are still whitelisted). If we could instead edit the title of the torrent page that would be a much easier solution.

4. torrent client whitelisting. this one isn't so important but it'd be a nice feature nonetheless. 

5. "download torrent" or "download magnet" shouldn't work if site is private. the torrent won't work anyway but it makes for a poor experience for a new user. We did it here https://github.com/rpdl-net/torrust/commit/aa264dbdf03302704e3064401cd493c5ce193c98 and here https://github.com/rpdl-net/torrust/commit/b713911c67269fe61c7ecfc38a20ffa34354bfa4 but they're bandaid patches.

6. our aim is to be as clueless about our users as possible. at the moment we replace the entire email column with random 10char strings every hour. if possible we'd like the option of not requiring an email at all upon sign up. it doesn't really bother us if people can't reset passwords etc. maybe something neat like a onetime code for their account that they could use to reset passwords? just spitballing really, if you can think of a better solution we'd be all ears.

7. from an admin point of view, we'd love to be able to track things like, amount of different IP addresses downloading per passkey so we can tell if people are leaking torrents and disable their accounts. 

8. tracking user statistics like amount uploaded & downloaded. again, not very important - it's pretty highly requested from our users though - nerding over stats is serious business lol.

---

that's all i can think of really. thank you so much for this project, it's amazing. we've submitted a few PRs from our fork (and we'll continue to do so if we think they're good for the main repo).

If you ever do choose to accept donations I'd like to send you something to get a case of beer!

**Comment by mickvandijke** [comment]
Hey @ldpr , thanks for using Torrust! It's good to hear it is working well for you and your users 😃 .

I agree with all the proposed features and I will include the following in the coming update of the torrust-index (v1.1.0):

- [x] 1. https://github.com/torrust/torrust/issues/40
- [x] 3. https://github.com/torrust/torrust/issues/41
- [x] 5. https://github.com/torrust/torrust/issues/42
- [x] 6. https://github.com/torrust/torrust/issues/32

Feature 2 will take more time, so It will probably not be included in the coming release, but it will be added in the minor release after the one we are working on now (so in v1.2.0).

4, 7 and 8 are torrust-tracker specific features, so I will make a new issue for those in the tracker repo.

Thank you for sharing your ideas and feedback! And I haven't had time yet to review the new PR's, but they are very much appreciated 😃 !

**Comment by da2ce7** [comment]
- https://github.com/torrust/torrust-index/discussions/309
